src/panels/config/devices/device-detail/ha-device-entities-card.ts

```diff
@@ -163,17 +163,27 @@ export class HaDeviceEntitiesCard extends LitElement {
     if (this.hass) {
       element.hass = this.hass;
       const stateObj = this.hass.states[entry.entity_id];
-      const name = stripPrefixFromEntityName(
-        computeStateName(stateObj),
-        this.deviceName.toLowerCase()
-      );
+
+      let name = entry.name
+        ? entry.name
```

Comment:
This should have been `stripPrefixFromEntityName(entry.name, this.deviceName.toLowerCase())`